### PR TITLE
Fix the upgrade migration of custom profile fields in #2545

### PR DESCRIPTION
**Pull request by nwsw**
The upgrade_2-1_postgresql.sql changes are untested, but do match the changes to upgrade_2-1_mysql.sql that are tested and confirmed to fix the upgrade issues in #2545.

Signed-off-by: Noteworthy Software, Inc online@noteworthysoftware.com

### CODE REVIEW

File: other/upgrade_2-1_mysql.sql [hunk 1]
```diff
@@ -850,13 +850,16 @@ INSERT INTO `{$db_prefix}custom_fields` (`col_name`, `field_name`, `field_desc`,
...
+// First see if we still have a gender column
+$results = $smcFunc['db_list_columns']('{db_prefix}members');
+if (in_array('gender', $results))
 {
 	$request = $smcFunc['db_query']('', '
 		SELECT id_member, aim, icq, msn, yim, location, gender
 		FROM {db_prefix}members');
 
 	$inserts = array();
+	$genderTypes = array(1=>"Male",2=>"Female");
```
Comment (Dragooon) [inline other/upgrade_2-1_mysql.sql]: This needs to be $genderTypes = array(1 => 'Male', 2 => 'Female');, we don't use double quotes for simple strings and have space between the assignment operator

File: other/upgrade_2-1_postgresql.sql [hunk 2]
```diff
@@ -907,13 +907,16 @@ INSERT INTO {$db_prefix}custom_fields (col_name, field_name, field_desc, field_t
...
+// First see if we still have a gender column
+$results = $smcFunc['db_list_columns']('{db_prefix}members');
+if (in_array('gender', $results))
 {
 	$request = $smcFunc['db_query']('', '
 		SELECT id_member, aim, icq, msn, yim, location, gender
 		FROM {db_prefix}members');
 
 	$inserts = array();
+	$genderTypes = array(1=>"Male",2=>"Female");
```
Comment (Dragooon) [inline other/upgrade_2-1_postgresql.sql]: Same as above

File: other/upgrade_2-1_mysql.sql [hunk 3]
```diff
@@ -850,13 +850,16 @@ INSERT INTO `{$db_prefix}custom_fields` (`col_name`, `field_name`, `field_desc`,
 ---# Converting member values...
 ---{
 // We cannot do this twice
-if (@$modSettings['smfVersion'] < '2.1')
```
Comment (Oldiesmann) [inline other/upgrade_2-1_mysql.sql:853]: If this is removed, the query to pull the IM fields, location and gender will fail if those columns aren't there. That's specifically why we have that check there.

Comment (nwsw) [inline other/upgrade_2-1_mysql.sql:853]: If you don't remove it, the included code will never be executed.

The other places that use this same condition are broken.for the same reason as described in #2545. The broken places are from the following:  dc573be2bc7565e3ba5c9b985115cc2931b3a0ed, 3e0417c56016bcae7859ccdbe93590ad88e1b25d, and af915c8fee045a0454233922556c94e5d67dbd9f.

If you do not uderstand why this is an issue, then you should review the following area of the upgrade.php script:

``` php
    // All possible files.
    // Name, <version, insert_on_complete
    $files = array(
        array('upgrade_1-0.sql', '1.1', '1.1 RC0'),
        array('upgrade_1-1.sql', '2.0', '2.0 a'),
        array('upgrade_2-0_' . $type . '.sql', '2.1', '2.1 dev0'),
        array('upgrade_2-1_' . $type . '.sql', '3.0', SMF_VERSION),
    );
```

Comment (Oldiesmann) [inline other/upgrade_2-1_mysql.sql:853]: I didn't know that was done. The easiest way to get around this issue is to only execute those particular sections of code if the version is 2.1 dev0.

Comment (Dragooon) [inline other/upgrade_2-1_mysql.sql:853]: Wouldn't checking for columns like it is being done in this PR be enough?

Comment (Oldiesmann) [inline other/upgrade_2-1_mysql.sql:853]: I didn't even notice that somehow. That should work for this at least. We'll still need to adjust anything else that checks for version < 2.1 in this file.

**Pull request merged**